docs/plugins/hook.rst

```diff
@@ -0,0 +1,77 @@
+Hook Plugin
+===============
+
+Internally, beets sends events to plugins when an action finishes. These can
+range from importing a song (``import``) to beets exiting (``cli_exit``), and
+provide a very flexible way to perform actions based on the events. This plugin
+allows you to run commands when an event is emitted by beets, such as syncing
+your library with another drive when the library is updated.
+
+Hooks are currently run in the order defined in the configuration, however this
+is dependent on beets itself (may change in the future) and cannot be controlled
+by this plugin.
+
+.. _hook-configuration:
+
+Configuration
+-------------
+
+To configure the plugin, make a ``hook:`` section in your configuration
+file. The available options are:
+
+- **hooks**: A list of events and the commands to run (see :ref:`individual-hook-configuration`)
+  Default: Empty.
+- **substitute_event**: The string to replace in each command with the name of
+  the event executing it. This can be used to allow one script to act
+  differently depending on the event it was called by. Can be individually
+  overridden (see :ref:`individual-hook-configuration`).
+  Default: ``%EVENT%``
+- **shell**: Run each command in a shell. Can be individually overridden (see
+  :ref:`individual-hook-configuration`).
+  Default: ``yes``
```

Comment:
Well, not using a shell by default would certainly be more secure! But if we bypass it entirely by default, then a configuration like this:

    hook:
      hooks:
        - event: cli_exit
          command: echo "have a nice day!"

won't work (as things currently stand). The plugin will try to find an executable named `echo "have a nice day!"`, which won't do!

We could, however, use `shlex`, as I hinted before, to parse the arguments. Then we'd get shell-like behavior for arguments formatted intuitively—without actually using a shell. If we do that, then IMO we should remove the `shell` option altogether; the `shlex` route would be the only option.